instructions.texi

```diff
@@ -1816,48 +1815,209 @@ Constants Vector: [ntchcdr (1550 1551 1552) 2]
+@verbatim
+PC  Byte  Instruction
+ 0  192   constant[0] (1590 1591)
+ 1  159   nreverse
+ 2  135   return
+
+Constants Vector: [(1590 1591)]
+@end verbatim
+
+@end table
 
 @node setcar
 @unnumberedsubsubsec @code{setcar} (160)
 @kindex setcar
 Call @code{setcar} with two stack arguments.
 
-Added in Emacs 19.34.
+@table @strong
+@item Implements:
+@code{S[1] <- (setcar S[1] TOS); top--}.
+@item Generated via:
+@code{setcar}.
+@item Instruction size:
+1 byte
+@item Stack effect:
+@math{-2+1}.
+@item Added in:
+Emacs 19.34.
+@item Example:
+With lexical binding in effect, @code{(defun setcar-eg(l) (setcar l 1600)))} generates:
+@c ((lexical . t) (optimize . nil))
+@verbatim
+PC  Byte  Instruction
+ 0  137   dup
+ 1  192   constant[0] 1600
+ 2  160   setcar
+ 3  135   return
+
+Constants Vector: [1600]
+@end verbatim
+
+@end table
 
 @node setcdr
 @unnumberedsubsubsec @code{setcdr} (161)
 @kindex setcdr
 Call @code{setcdr} with two stack arguments.
 
-Added in Emacs 19.34.
+@table @strong
+@item Implements:
+@code{S[1] <- (setcdr S[1] TOS); top--}.
+@item Generated via:
+@code{setcdr}.
+@item Instruction size:
+1 byte
+@item Stack effect:
+@math{-2+1}.
+@item Added in:
+Emacs 19.34.
+@item Example:
+With lexical binding in effect, @code{(defun setcdr-eg(l) (setcdr l 1610)))} generates:
+@c ((lexical . t) (optimize . nil))
+@verbatim
+PC  Byte  Instruction
+ 0  137   dup
+ 1  192   constant[0] 1610
+ 2  161   setcdr
+ 3  135   return
+
+Constants Vector: [1610]
+@end verbatim
+
+@end table
 
 @node car-safe
 @unnumberedsubsubsec @code{car-safe} (162)
 @kindex car-safe
+
 Call @code{car-safe} with one argument.
 
-Added in Emacs 19.34.
+@table @strong
+@item Implements:
+@code{TOS <- (car-safe TOS)}.
+@item Generated via:
+@code{car-safe}.
+@item Instruction size:
+1 byte
+@item Stack effect:
+@math{-1+1}.
+@item Added in:
+Emacs 19.34.
+@item Example:
+With lexical binding in effect, @code{(defun car-safe-eg(l) (car-safe l))} generates:
+@c ((lexical . t) (optimize . nil))
+@verbatim
+PC  Byte  Instruction
+ 0  137   dup
+ 1  162   car-safe
+ 2  135   return
+@end verbatim
+
+@end table
 
 @node cdr-safe
 @unnumberedsubsubsec @code{cdr-safe} (163)
 @kindex cdr-safe
-Call @code{cdr-safe} with one argument.
+Call @code{cdr-safe} with one stack argument.
+
+@table @strong
+@item Implements:
+@code{TOS <- (cdr-safe TOS)}.
+@item Generated via:
+@code{cdr-safe}.
+@item Instruction size:
+1 byte
+@item Stack effect:
+@math{-1+1}.
+@item Added in:
+Emacs 19.34.
+@item Example:
+With lexical binding in effect, @code{(defun cdr-safe-eg(l) (cdr-safe l))} generates:
+@c ((lexical . t) (optimize . nil))
+@verbatim
+PC  Byte  Instruction
+ 0  137   dup
+ 1  163   cdr-safe
+ 2  135   return
+@end verbatim
+
+@end table
 
-Added in Emacs 19.34.
 @node nconc
 @unnumberedsubsubsec @code{nconc} (164)
 @kindex nconc
 Call @code{nconc} with two stack arguments.
 
-Added in Emacs 19.34.
+@table @strong
+@item Implements:
+@code{S[1] <- (nconc S[1] TOS; top--)}.
```

Comment:
You sometimes have the `top--` part in the parens, and sometimes after them. Which one is the official version, so I can fix the others?